src/x/serialize/decoder.go

```diff
@@ -237,3 +239,17 @@ func (d *decoder) Duplicate() ident.TagIterator {
 	}
 	return iter
 }
+
+func (d *decoder) Hash() (uint32, error) {
+	if d.checkedData == nil {
+		return 0, nil
+	}
+	if d.digest == 0 {
+		digest := adler32.New()
+		if _, err := digest.Write(d.checkedData.Bytes()); err != nil {
```

Comment:
Unfortunately won't be able to depend on this being equivalent to other Hash methods on other tag iterators, if we want to depend on equality with another set of tags (that could be a non-encoded set of tags) we need to generate the hash the same way, rolling hash of name+value in order that tags appear).